ui/Makefile

```diff
@@ -32,18 +33,23 @@ GOPATH_BIN := $(GOPATH)/bin
 GO_BINDATA := $(GOPATH_BIN)/go-bindata
 
 TYPESCRIPT_TARGET      := js/app.js
+TEST_TARGET            := js/test.js
 GOBINDATA_SOURCES      := $(TYPESCRIPT_TARGET) index.html $(shell find css js/libs -type f)
 GOBINDATA_TARGET       := ./embedded.go
 GOBINDATA_DEBUG_TARGET := ./embedded_debug.go
 
 .PHONY:
-all: $(GOBINDATA_TARGET)
+all: $(GOBINDATA_TARGET) $(TEST_TARGET)
 	rm -f $(GOBINDATA_DEBUG_TARGET)
 
 .PHONY:
 debug: $(GOBINDATA_DEBUG_TARGET)
 	rm -f $(GOBINDATA_TARGET)
 
+.PHONY:
+test: $(TEST_TARGET)
```

Comment:
Eh, I thought it was transitively depending on npm.installed through TEST_TARGET, but you're right that might not work if `make test` is the first thing you do.